frontend/src/api/openai.ts

```diff
@@ -48,47 +50,48 @@ emoji: 🎉
 			content: sp
 		}
 	]
-
-	if (image) {
-		// TODO: configurable
-		if (model.startsWith('gpt')) {
-			model = 'gpt-4-vision-preview'
-		}
-		let imageUrl = image
-		// OpenAI wants a data url, ollama just wants base64 bytes
-		if (model.startsWith('ollama/')) {
-			const parts = image.toString().split(',')
-			imageUrl = parts.pop() ?? ''
-		}
-		messages.push({
-			role: 'user',
-			content: [
-				{
-					type: 'text',
-					text: 'This is a screenshot of a web component I want to replicate.  Please generate HTML for it.'
-				},
-				{
-					type: 'image_url',
-					image_url: {
-						url: imageUrl
+	if (action === 'create') {
+		// Call the vision models only for creating action
+		if (image) {
+			// TODO: configurable
+			if (model.startsWith('gpt')) {
+				model = 'gpt-4-vision-preview'
+			}
+			let imageUrl = image
+			// OpenAI wants a data url, ollama just wants base64 bytes
+			if (model.startsWith('ollama/')) {
+				const parts = image.toString().split(',')
+				imageUrl = parts.pop() ?? ''
+			}
+			const textImageRequirements = query ?
+				`The following are some special requirements: \n ${query}`
+				: '';
+			messages.push({
+				role: 'user',
+				content: [
+					{
+						type: 'text',
+						text: `This is a screenshot of a web component I want to replicate.  Please generate HTML for it.\n ${textImageRequirements}`
+					},
+					{
+						type: 'image_url',
+						image_url: {
+							url: imageUrl
+						}
 					}
-				}
-			]
-		})
-	}
-
-	if (html === undefined || html === '') {
-		if (query !== 'image-upload') {
+				]
+			})
+		} else { 
 			messages.push({
 				role: 'user',
 				content: query
 			})
 		}
 	} else {
-		let userPrompt = 'Address the FIX comments.'
-		if (query !== 'html-comments-only') {
-			userPrompt = query
-		}
+		// Annotation comments should like <!--FIX (1): make the image larger-->
+		const hasAnnotationComments = /<!--FIX (\(\d+\)): (.+)-->/g.test(html as string);
+		let userPrompt = hasAnnotationComments ? 'Address the FIX comments.': query
```

Comment:
I think this logic might be problematic.  Imagine the following scenario:

1. User generates HTML
2. User annotates with a fix comment
3. User wants to iterate again with a query comment

The HTML that's coming in at #3 above might still have <!--FIX comments in it.  I've definitely seen the LLM return new HTML that still has the FIX comments.  I know I have logic somewhere to just remove comments from the HTML which we could do on ingest and this might be ok...  I'll give it a try locally and see if it's a problem.

Reply:
Thanks for the heads up.
Remove the FIX comments from the LLM response might be a solution. But I obviously don't know as much about this as you do, so if you think it works, please tell where it should be changed or you make the change directly.